### ISSUE TITLE
Feature request : implement double opt-in

### ISSUE THREAD

**Issue by MagicFab** [pentacent/keila#141]
I'd like to suggest a default feature found in other similar software : an option to request a new email subscriber to confirm their email is valid by clicking on a link sent in the first confirmation email after subscription. This is commonly known as double opt-in.
This improves a lot the quality of email subscribers by minimizing typing mistakes while also reducing SPAM, specially when not using the CAPTCHA method.

This could be an optional, but checked-by-default checkbox under the option "Require captcha" option at form creation.

https://www.campaignmonitor.com/resources/glossary/double-opt-in/

**Comment by wmnnd** [comment]
Thanks for creating this issue. You’re right, double-opt is an absolute must and I’ll add it to the roadmap.

I wonder whether this is best done via configuring the form (i.e. you set the form to double-opt-in and then also configure the message you’d like to send) - or if this should be part of the automation feature #132.

Happy to hear everyone’s thoughts!

**Comment by victorbjorklund** [comment]
Agreed. This is a must have. Also would be nice to have access to it when adding with the api (if you use your own form instead of the form in keila).

**Comment by wmnnd** [comment]
This is how I think the double-opt in feature should work:

- Add a new tab to the form editor called "Double-opt-in" with the following elements:
  - Checkbox: Enable double-opt-in
  - Select: Sender
  - Select: Email template
  - Text field: Email subject
  - Text area: Markdown editor for editing the opt-in content (no WYSIWYG for the first iteration)
    -  This field validates the inclusion of the `{{ opt_in_link }}` variable.
  - Text field: Redirect URL
- Contacts that sign up from a form with double-opt-in should be stored in a separate table, ideally with their email address encrypted/obfuscated.
- Once the opt-in link is clicked, a regular contact is created

**Comment by wmnnd** [comment]
This feature was implemented in #219 and is available in Keila 0.13.0  :partying_face:
